src/pages/Account/Admin.tsx

```diff
@@ -5,61 +5,192 @@ import { useUser } from "hooks/useUser";
 import AuthenticationModal from "components/organisms/AuthenticationModal";
 import AdminEvent from "./AdminEvent";
 import WithNavigationBar from "components/organisms/WithNavigationBar";
-import { useHistory, Link } from "react-router-dom";
+import {
+  Link,
+  useParams,
+  Switch,
+  Route,
+  useRouteMatch,
+  useLocation,
+} from "react-router-dom";
 import { useFirestoreConnect } from "react-redux-firebase";
-import { useSelector } from "hooks/useSelector";
+import { useSelector, useKeyedSelector } from "hooks/useSelector";
+import { Venue } from "types/Venue";
+import { WithId } from "utils/id";
+import { createUrlSafeName } from "api/admin";
 
-const Admin: React.FC = () => {
-  const { user } = useUser();
-  const { push } = useHistory();
-  const [showCreateEventModal, setShowCreateEventModal] = useState(false);
+type VenueListProps = {
+  selectedVenueId?: string;
+};
+
+const VenueList: React.FC<VenueListProps> = ({ selectedVenueId }) => {
   const venues = useSelector((state) => state.firestore.ordered.venues);
+  return (
+    <>
+      <h4>My Venues</h4>
+      <ul>
+        {venues?.map((venue) => (
+          <li
+            key={venue.id}
+            style={venue.id === selectedVenueId ? { color: "red " } : {}}
+          >
+            <Link to={`/admin/venue/${venue.id}`}>{venue.name}</Link>
+          </li>
+        ))}
+      </ul>
+      <div className="centered-flex">
+        <Link className="btn btn-primary" to="/admin/venue/creation">
+          Create a venue
+        </Link>
+      </div>
+    </>
+  );
+};
+
+type VenueDetailsProps = {
+  venueId: string;
+};
+
+type VenueDetailsPartProps = {
+  venue: WithId<Venue>;
+};
 
+const VenueDetails: React.FC<VenueDetailsProps> = ({ venueId }) => {
+  const match = useRouteMatch();
+  const location = useLocation();
+  const { venues } = useKeyedSelector(
+    (state) => ({
+      venues: state.firestore.data.venues ?? {},
+    }),
+    ["venues"]
+  );
+
+  const venue = venues[venueId];
+
+  if (!venue) {
+    return <>Oops, seems we can't find your venue!</>;
+  }
+
+  return (
+    <>
+      <ul>
+        {[
+          { url: `${match.url}`, label: "Venue Infos" },
+          { url: `${match.url}/appearance`, label: "Appearance" },
+          { url: `${match.url}/events`, label: "Events" },
+        ].map((l) => (
```

Comment:
I don't like this variable name. could we name it tab?

Reply:
fixed